cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,177 @@
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+// RunVolumeInfo runs info command and make call to DisplayVolumeInfo
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		return nil
+	}
+	if annotation == nil {
+		fmt.Println("annotation is nil")
+		return nil
+	}
+	if annotation.ControllerStatus != "Running" {
+		fmt.Println("Volume not reachable, ", annotation.ControllerStatus)
+		return nil
+	}
+	// Initiallize an instance of ReplicaCollection, json response recieved from the
+	// controllerIP:9501/v1/replicas is to be parsed into this structure via GetVolumeStats.
+	// An API needs to be passed as argument.
+	collection := client.ReplicaCollection{}
+	controllerClient := client.ControllerClient{}
+	_, err = controllerClient.GetVolumeStats(annotation.ClusterIP+v1.ControllerPort, v1.InfoAPI, &collection)
+	if err != nil {
+		return err
+	}
+
+	c.DisplayVolumeInfo(annotation, collection)
+	return nil
+}
+
+// DisplayVolumeInfo displays the outputs in standard I/O.
+// Currently It displays volume access modes and target portal details only.
+func (c *CmdVolumeInfoOptions) DisplayVolumeInfo(a *Annotations, collection client.ReplicaCollection) error {
+	var (
+		// address, mode are used here as blackbox for the replica info
+		// address keeps the ip and access mode details respectively.
+		address, mode []string
+		replicaCount  int
+		portalInfo    PortalInfo
+	)
+	const (
+		replicaTemplate = `
+================= Replica Details =================
+IP            AccessMode               Status
+{{range $key, $value := .}}
+{{$value.IP}}     {{$value.AccessMode}}                       {{$value.Status}}
+{{end}}
+===================================================
+`
+		portalTemplate = `
+================= Portal Details ==================
+IQN     :   {{.IQN}}
+Volume  :   {{.VolumeName}}
+Portal  :   {{.Portal}}
+Size    :   {{.Size}}
+Status  :   {{.Status}}
+===================================================
+`
+	)
+	portalInfo = PortalInfo{
+		a.Iqn,
+		c.volName,
+		a.TargetPortal,
+		a.VolSize,
+		a.ControllerStatus,
+	}
+	tmpl, err := template.New("VolumeInfo").Parse(portalTemplate)
+	if err != nil {
+		fmt.Println("Can't Parse the template, found error :", err)
+		return nil
+	}
+	err = tmpl.Execute(os.Stdout, portalInfo)
+	if err != nil {
+		fmt.Println("Can't execute the template, found error :", err)
+		return nil
+	}
+	replicaCount, _ = strconv.Atoi(a.ReplicaCount)
+	length := len(collection.Data)
+	// This case will occur only if user has manually specified zero replica.
+	if replicaCount == 0 || length == 0 {
+		fmt.Println("No Replicas are running, please check the volume's pod specification or try again later.")
```

Comment:
The error message could be changed to "None of the replicas are online. Please check the status of the replica's using: kubectl get pods --all-namespaces | grep <volname>" 

Will the above work.